test/GasAccounting.t.sol

```diff
@@ -99,307 +124,515 @@ contract MockGasAccounting is GasAccounting, Test {
     function calldataLengthPremium() external pure returns (uint256) {
         return _CALLDATA_LENGTH_PREMIUM;
     }
+
+    function solverBaseGasUsed() external pure returns (uint256) {
+        return _SOLVER_BASE_GAS_USED;
+    }
+
+    function activeEnvironment() public view returns (address) {
+        return _activeEnvironment();
+    }
 }
 
-contract GasAccountingTest is Test {
+contract GasAccountingTest is BaseTest {
     MockGasAccounting public mockGasAccounting;
-    address executionEnvironment = makeAddr("executionEnvironment");
-
     uint256 gasMarker;
     uint256 initialClaims;
     SolverOperation solverOp;
+    address executionEnvironment;
+    uint256 snapshotId;
+
+    function setUp() public override {
+        // Run the base setup
+        super.setUp();
+
+        // Compute expected addresses for the deployment
+        address expectedAtlasAddr = vm.computeCreateAddress(payee, vm.getNonce(payee) + 1);
+        bytes32 salt = keccak256(abi.encodePacked(block.chainid, expectedAtlasAddr, "AtlasFactory 1.0"));
+        ExecutionEnvironment execEnvTemplate = new ExecutionEnvironment{ salt: salt }(expectedAtlasAddr);
+
+        // Initialize MockGasAccounting
+        mockGasAccounting = new MockGasAccounting(
+            DEFAULT_ESCROW_DURATION, address(atlasVerification), address(simulator), payee, address(execEnvTemplate)
+        );
+
+        // Initialize TestAtlas storage slots
+        initializeTestAtlasSlots();
 
-    function setUp() public {
-        mockGasAccounting = new MockGasAccounting(0, address(0), address(0), address(0));
-        uint256 _gasMarker = gasleft();
+        gasMarker = gasleft();
+        mockGasAccounting.initializeLock{ value: 0 }(solverOneEOA, gasMarker);
+        initialClaims = getInitialClaims(gasMarker);
+        solverOp.from = solverOneEOA; // Use the solverOneEOA address from BaseTest
+        solverOp.to = solverOneEOA;
+        solverOp.data = abi.encodePacked("calldata");
+        executionEnvironment = mockGasAccounting.activeEnvironment();
+
+        // Ensure the execution environment starts with zero balance
+        deal(executionEnvironment, 0);
 
-        mockGasAccounting.initializeLock(executionEnvironment, _gasMarker, 0);
+        // Take a snapshot before each test
+        snapshotId = vm.snapshot();
```

Comment:
So this snapshot thing is only really needed when you want to revert to a previous state within the same test.

Reply:
resolved